### ISSUE TITLE
Fix multiple versioned pickle wrappers

### ISSUE THREAD

**Issue by corranwebster** [enthought/apptools#133]
Apptools provides the `sweet_pickle` subpackage to handle versioning of pickled objects to allow unpickling of objects after changes to the class structure.

The `persistence` subpackage has a slightly different focus: it is more about extracting state as a dictionary, but it has similar themes.

Mayavi uses `persistence`, but it also uses `apptools.naming` which internally uses `sweet_pickle`, so we cannot simply deprecate `sweet_pickle`.

**Comment by aaronayres35** [comment]
`persistence` coverage:

running the tests under coverage in `apptools/apptools/persistence/tests` gives:

<img width="1014" alt="Screen Shot 2020-11-10 at 11 05 18 AM" src="https://user-images.githubusercontent.com/36972686/98707164-57578080-2345-11eb-946d-cab26520a0e4.png">


Running the full testsuite for apptools under coverage from the apptools root directory gives:

<img width="1092" alt="Screen Shot 2020-11-10 at 11 08 24 AM" src="https://user-images.githubusercontent.com/36972686/98707250-6f2f0480-2345-11eb-98c3-b2cecf17410a.png">

**Comment by aaronayres35** [comment]
The following are all the uses of `sweet_pickle` in naming: 
https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/naming/object_serializer.py#L25

https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/naming/object_serializer.py#L59

https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/naming/object_serializer.py#L89

Looking at `sweet_pickle.load` and `sweet_pickle.dump` more closely:
https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/sweet_pickle/__init__.py#L112-L113

The `Unpickler` here is `apptools.sweet_pickle.versioned_unpickler.VersionedUnpickler` which inherits the `load` method from `NewUnpickler` here:
https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/sweet_pickle/versioned_unpickler.py#L112-L131

Note this is identical to the same method on the `apptools.persistence.versioned_unpickler.VersionUnpickler` class which inherits the `load` method from `VersionedUnpickler` here:
https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/persistence/versioned_unpickler.py#L37-L56


So it seems this could easily be replaced to use `persistence` instead.  

Likewise, looking closer at dump:

https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/sweet_pickle/__init__.py#L126-L131

It seems the only thing `sweet_pickle` provides over `six.moves.cPickle` is the `_flush_traits` function:
https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/sweet_pickle/__init__.py#L161-L168
However, this docstring for a function of the same name in `persistence.StatePickler` suggests this isn't needed:
https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/persistence/state_pickler.py#L316-L321

There are also other cases where `apptools.naming` imports: `import six.moves.cPickle as pickle`.


Overall it seems to me that `sweet_pickle` can very easily be replaced by `persistence` / `six.moves.cPickle` in `naming`.  This is the only use of `sweet_pickle` inside `apptools` aside from the `apptools.sweet_pickle` subpackage itself.  Additionally, it seems `sweet_pickle` is not used by any other active projects.  With that it seems we should be fine to make the replacement in `naming` and then kill `sweet_pickle` altogether?

**Comment by aaronayres35** [comment]
Looking at using `persistence` instead of `sweet_pickle` to call `load`
> So it seems this could easily be replaced to use `persistence` instead.

I tried this and there was actually one other minor problem.  In persistence, `NewUnpickler` subclasses `Unpickler` which comes from `from pickle import *`.  On the other hand, `sweet_pickle` does the following:
https://github.com/enthought/apptools/blob/a693c79e8547ccaafcf89f230c0b17edbc780655/apptools/sweet_pickle/versioned_unpickler.py#L25-L28

As a result, when running the `naming` test suite there is a failure with the traceback that ends in:
```
File "/Users/aayres/Desktop/apptools/apptools/persistence/versioned_unpickler.py", line 46, in load
    dispatch = self.dispatch
AttributeError: 'VersionedUnpickler' object has no attribute 'dispatch'
```


Since we are soon to drop support for python 2, it seems we will always want `_Unpickler` here anyway, so I went ahead and added `from pickle import _Unpickler as Unpickler` at the top of the `persistence.unversioned_pickler.py` file and after that change all the tests pass.
